libs/java/server_common/src/main/java/com/yahoo/athenz/common/server/log/jetty/JettyConnectionLogger.java

```diff
@@ -131,8 +131,12 @@ synchronized ConnectionInfo setSslHandshakeFailure(Throwable exception) {
             this.sslHandshakeFailureException = sslHandshakeException.getClass().getName();
             this.sslHandshakeFailureMessage = sslHandshakeException.getMessage();
             // If the error isn't clear, try to get it from the exception inner cause
-            if (exception.getCause() != null && GENERAL_SSL_ERROR.equals(this.sslHandshakeFailureMessage)) {
+            if (exception.getCause() != null) {
                 this.sslHandshakeFailureCause = getInnerCause(exception, sslHandshakeException.getMessage());
+                // If the cause is identical to the message, no need to print it so we'll set it to null
+                if (this.sslHandshakeFailureCause.equals(this.sslHandshakeFailureMessage)) {
```

Comment:
are we absolutely sure that this.sslHandshakeFailureCause will never be null? we might be want be safe and do proper checks?

Reply:
> are we absolutely sure that this.sslHandshakeFailureCause will never be null? we might be want be safe and do proper checks?

That would mean that one of the Jetty exceptions has a null message which I don't think is likely but I added the check to be on the safe side.